lib/matplotlib/projections/geo.py

```diff
@@ -269,24 +269,23 @@ def __init__(self, resolution):
             self._resolution = resolution
 
         def transform_non_affine(self, ll):
-            longitude = ll[:, 0:1]
-            latitude  = ll[:, 1:2]
+            longitude = ll[:, 0]
+            latitude = ll[:, 1]
 
             # Pre-compute some values
             half_long = longitude / 2.0
             cos_latitude = np.cos(latitude)
 
             alpha = np.arccos(cos_latitude * np.cos(half_long))
-            # Mask this array or we'll get divide-by-zero errors
-            alpha = ma.masked_where(alpha == 0.0, alpha)
-            # The numerators also need to be masked so that masked
-            # division will be invoked.
+            # Avoid divide-by-zero errors using same method as NumPy.
+            alpha[alpha == 0.0] = 1e-20
             # We want unnormalized sinc.  numpy.sinc gives us normalized
-            sinc_alpha = ma.sin(alpha) / alpha
+            sinc_alpha = np.sin(alpha) / alpha
 
-            x = (cos_latitude * ma.sin(half_long)) / sinc_alpha
-            y = (ma.sin(latitude) / sinc_alpha)
-            return np.concatenate((x.filled(0), y.filled(0)), 1)
+            xy = np.empty_like(ll, float)
```

Comment:
I like column_stack, which I think has a very descriptive name... (but it's just personal preference)

Reply:
pre-allocating and then assigning to slices is faster, and still very readable.

Reply:
I think it's essentially the same speed (not that it really matters):
```
In [15]: %%timeit x = np.random.rand(10000); y = np.random.rand(10000)
    ...: z = np.column_stack([x, y])
    ...: 
37.8 µs ± 216 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [16]: %%timeit x = np.random.rand(10000); y = np.random.rand(10000)
    ...: z = np.empty((10000, 2))
    ...: z[:, 0] = x; z[:, 1] = y
    ...: 
36.7 µs ± 1.39 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```